File: src/Application Specific/Classes/Entity Collections/Jobs/Job/Job.vb
Namespace Entity

    Namespace Jobs

        Public Class Job

            Private _dataTypeValidator As DataTypeValidator

            Public Sub New()
                _dataTypeValidator = New DataTypeValidator
            End Sub

#Region "Class Properties"

            Public Property IgnoreExceptionsOnSetMethods() As Boolean
                Get
                    Return Me._dataTypeValidator.IgnoreExceptionsOnSetMethods
                End Get
                Set(ByVal Value As Boolean)
                    Me._dataTypeValidator.IgnoreExceptionsOnSetMethods = Value
                End Set
            End Property

            Public ReadOnly Property Errors() As Hashtable
                Get
                    Return _dataTypeValidator.Errors
                End Get
            End Property

            Public ReadOnly Property DTValidator() As DataTypeValidator
                Get
                    Return _dataTypeValidator
                End Get
            End Property

            Private _exists As Boolean = False

            Public Property Exists() As Boolean
                Get
                    Return _exists
                End Get
                Set(ByVal Value As Boolean)
                    _exists = Value
                End Set
            End Property

            Private _deleted As Boolean = False

            Public ReadOnly Property Deleted() As Boolean
                Get
                    Return _deleted
                End Get
            End Property

            Public WriteOnly Property SetDeleted() As Boolean
                Set(ByVal Value As Boolean)
                    _deleted = Value
                End Set
            End Property

#End Region

#Region "Job Properties"

            Private _JobID As Integer = 0

            Public ReadOnly Property JobID() As Integer
                Get
                    Return _JobID
                End Get
            End Property

            Public WriteOnly Property SetJobID() As Object
                Set(ByVal Value As Object)
                    _dataTypeValidator.SetValue(Me, "_JobID", Value)
                End Set
            End Property

            Private _PropertyID As Integer = 0

            Public ReadOnly Property PropertyID() As Integer
                Get
                    Return _PropertyID
                End Get
            End Property

            Public WriteOnly Property SetPropertyID() As Object
                Set(ByVal Value As Object)
                    _dataTypeValidator.SetValue(Me, "_PropertyID", Value)
                End Set
            End Property

            Private _JobDefinitionEnumID As Integer = 0

            Public ReadOnly Property JobDefinitionEnumID() As Integer
                Get
                    Return _JobDefinitionEnumID
                End Get
            End Property

            Public WriteOnly Property SetJobDefinitionEnumID() As Object
                Set(ByVal Value As Object)
                    _dataTypeValidator.SetValue(Me, "_JobDefinitionEnumID", Value)
                End Set
            End Property

            Private _JobTypeID As Integer = 0

            Public ReadOnly Property JobTypeID() As Integer
                Get
                    Return _JobTypeID
                End Get
            End Property

            Public WriteOnly Property SetJobTypeID() As Object
                Set(ByVal Value As Object)
                    _dataTypeValidator.SetValue(Me, "_JobTypeID", Value)
                End Set
            End Property

            Private _StatusEnumID As Integer = 0

            Public ReadOnly Property StatusEnumID() As Integer
                Get
                    Return _StatusEnumID
                End Get
            End Property

            Public WriteOnly Property SetStatusEnumID() As Object
                Set(ByVal Value As Object)
                    _dataTypeValidator.SetValue(Me, "_StatusEnumID", Value)
                End Set
            End Property

            Private _DateCreated As DateTime = Nothing

            Public Property DateCreated() As DateTime
                Get
                    Return _DateCreated
                End Get
                Set(ByVal Value As DateTime)
                    _DateCreated = Value
                End Set
            End Property

            Private _CreatedByUserID As Integer = 0

            Public ReadOnly Property CreatedByUserID() As Integer
                Get
                    Return _CreatedByUserID
                End Get
            End Property

            Public WriteOnly Property SetCreatedByUserID() As Object
                Set(ByVal Value As Object)
                    _dataTypeValidator.SetValue(Me, "_CreatedByUserID", Value)
                End Set
            End Property

            Private _JobNumber As String = String.Empty

            Public ReadOnly Property JobNumber() As String
                Get
                    Return _JobNumber
                End Get
            End Property

            Public WriteOnly Property SetJobNumber() As Object
                Set(ByVal Value As Object)
                    _dataTypeValidator.SetValue(Me, "_JobNumber", Value)
                End Set
            End Property

            Private _PONumber As String = String.Empty

            Public ReadOnly Property PONumber() As String
                Get
                    Return _PONumber
                End Get
            End Property

            Public WriteOnly Property SetPONumber() As Object
                Set(ByVal Value As Object)
                    _dataTypeValidator.SetValue(Me, "_PONumber", Value)
                End Set
            End Property

            Private _QuoteID As Integer = 0

            Public ReadOnly Property QuoteID() As Integer
                Get
                    Return _QuoteID
                End Get
            End Property

            Public WriteOnly Property SetQuoteID() As Object
                Set(ByVal Value As Object)
                    _dataTypeValidator.SetValue(Me, "_QuoteID", Value)
                End Set
            End Property

            Private _ContractID As Integer = 0

            Public ReadOnly Property ContractID() As Integer
                Get
                    Return _ContractID
                End Get
            End Property

            Public WriteOnly Property SetContractID() As Object
                Set(ByVal Value As Object)
                    _dataTypeValidator.SetValue(Me, "_ContractID", Value)
                End Set
            End Property

            Private _ToBePartPaid As Boolean = False

            Public ReadOnly Property ToBePartPaid() As Boolean
                Get
                    Return _ToBePartPaid
                End Get
            End Property

            Public WriteOnly Property SetToBePartPaid() As Object
                Set(ByVal Value As Object)
                    _dataTypeValidator.SetValue(Me, "_ToBePartPaid", Value)
                End Set
            End Property

            Private _retention As Double = 0

            Public ReadOnly Property Retention() As Double
                Get
                    Return _retention
                End Get
            End Property

            Public WriteOnly Property SetRetention() As Object
                Set(ByVal Value As Object)
                    _dataTypeValidator.SetValue(Me, "_retention", Value)
                End Set
            End Property

            Private _CollectPayment As Boolean = 0

            Public ReadOnly Property CollectPayment() As Boolean
                Get
                    Return _CollectPayment
                End Get
            End Property

            Public WriteOnly Property SetCollectPayment() As Object
                Set(ByVal Value As Object)
                    _dataTypeValidator.SetValue(Me, "_CollectPayment", Value)
                End Set
            End Property

            Private _POC As Boolean = False

            Public ReadOnly Property POC() As Boolean
                Get
                    Return _POC
                End Get
            End Property

            Public WriteOnly Property SetPOC() As Object
                Set(ByVal Value As Object)
                    _dataTypeValidator.SetValue(Me, "_POC", Value)
                End Set
            End Property

            Private _OTI As Boolean = False

            Public ReadOnly Property OTI() As Boolean
                Get
                    Return _OTI
                End Get
            End Property

            Public WriteOnly Property SetOTI() As Object
                Set(ByVal Value As Object)
                    _dataTypeValidator.SetValue(Me, "_OTI", Value)
                End Set
            End Property

            Private _fOC As Boolean = False

            Public ReadOnly Property FOC() As Boolean
                Get
                    Return _fOC
                End Get
            End Property

            Public WriteOnly Property SetFOC() As Object
                Set(ByVal Value As Object)
                    _dataTypeValidator.SetValue(Me, "_fOC", Value)
                End Set
            End Property

            Private _salesRepUserID As Integer = 0

            Public ReadOnly Property SalesRepUserID() As Integer
                Get
                    Return _salesRepUserID
                End Get
            End Property

            Public WriteOnly Property SetSalesRepUserID() As Object
                Set(ByVal Value As Object)
                    _dataTypeValidator.SetValue(Me, "_salesRepUserID", Value)
                End Set
            End Property

            Private _jobCreationType As Integer = 0

            Public ReadOnly Property JobCreationType() As Integer
                Get
                    Return _jobCreationType
                End Get
            End Property

            Public WriteOnly Property SetJobCreationType() As Object
                Set(ByVal Value As Object)
                    _dataTypeValidator.SetValue(Me, "_jobCreationType", Value)
                End Set
            End Property

            Private _Headline As String = String.Empty

            Public ReadOnly Property Headline() As String
                Get
                    Return _Headline
                End Get
            End Property

            Public WriteOnly Property SetHeadline() As Object
                Set(ByVal Value As Object)
                    _dataTypeValidator.SetValue(Me, "_Headline", Value)
                End Set
            End Property

#End Region

#Region "Additional Properties"

            Private _assets As New ArrayList

            Public Property Assets() As ArrayList
                Get
                    Return _assets
                End Get
                Set(ByVal Value As ArrayList)
                    _assets = Value
                End Set
            End Property

            Private _jobOfWorks As New ArrayList

            Public Property JobOfWorks() As ArrayList
                Get
                    Return _jobOfWorks
                End Get
                Set(ByVal Value As ArrayList)
                    _jobOfWorks = Value
                End Set
            End Property

            Private _JobQualificationsLevels As New DataView

            Public Property JobQualificationsLevels() As DataView
                Get
                    Return _JobQualificationsLevels
                End Get
                Set(ByVal Value As DataView)
                    _JobQualificationsLevels = Value
                    _JobQualificationsLevels.AllowEdit = True
                    _JobQualificationsLevels.AllowNew = False
                    _JobQualificationsLevels.AllowDelete = False
                End Set
            End Property

            Private _JobSheets As New DataView

            Public Property JobSheets() As DataView
                Get
                    Return _JobSheets
                End Get
                Set(ByVal Value As DataView)
                    _JobSheets = Value
                    _JobSheets.AllowEdit = True
                    _JobSheets.AllowNew = False
                    _JobSheets.AllowDelete = False
                End Set
            End Property

#End Region

        End Class

    End Namespace

End Namespace

File: apollo/apollo.fsm/src/Application Specific/Classes/Entity Collections/Jobs/Job/Job.cs
using System;
using System.Collections;
using System.Data;
using Microsoft.VisualBasic.CompilerServices;

namespace FSM.Entity
{
    namespace Jobs
    {
        public class Job
        {
            private DataTypeValidator _dataTypeValidator;

            public Job()
            {
                _dataTypeValidator = new DataTypeValidator();
            }


            public bool IgnoreExceptionsOnSetMethods
            {
                get
                {
                    return _dataTypeValidator.IgnoreExceptionsOnSetMethods;
                }

                set
                {
                    _dataTypeValidator.IgnoreExceptionsOnSetMethods = value;
                }
            }

            public Hashtable Errors
            {
                get
                {
                    return _dataTypeValidator.Errors;
                }
            }

            public DataTypeValidator DTValidator
            {
                get
                {
                    return _dataTypeValidator;
                }
            }

            private bool _exists = false;

            public bool Exists
            {
                get
                {
                    return _exists;
                }

                set
                {
                    _exists = value;
                }
            }

            private bool _deleted = false;

            public bool Deleted
            {
                get
                {
                    return _deleted;
                }
            }

            public bool SetDeleted
            {
                set
                {
                    _deleted = value;
                }
            }



            private int _JobID = 0;

            public int JobID
            {
                get
                {
                    return _JobID;
                }
            }

            public object SetJobID
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_JobID", value);
                }
            }

            private int _PropertyID = 0;

            public int PropertyID
            {
                get
                {
                    return _PropertyID;
                }
            }

            public object SetPropertyID
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_PropertyID", value);
                }
            }

            private int _JobDefinitionEnumID = 0;

            public int JobDefinitionEnumID
            {
                get
                {
                    return _JobDefinitionEnumID;
                }
            }

            public object SetJobDefinitionEnumID
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_JobDefinitionEnumID", value);
                }
            }

            private int _JobTypeID = 0;

            public int JobTypeID
            {
                get
                {
                    return _JobTypeID;
                }
            }

            public object SetJobTypeID
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_JobTypeID", value);
                }
            }

            private int _StatusEnumID = 0;

            public int StatusEnumID
            {
                get
                {
                    return _StatusEnumID;
                }
            }

            public object SetStatusEnumID
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_StatusEnumID", value);
                }
            }

            private DateTime _DateCreated = default;

            public DateTime DateCreated
            {
                get
                {
                    return _DateCreated;
                }

                set
                {
                    _DateCreated = value;
                }
            }

            private int _CreatedByUserID = 0;

            public int CreatedByUserID
            {
                get
                {
                    return _CreatedByUserID;
                }
            }

            public object SetCreatedByUserID
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_CreatedByUserID", value);
                }
            }

            private string _JobNumber = string.Empty;

            public string JobNumber
            {
                get
                {
                    return _JobNumber;
                }
            }

            public object SetJobNumber
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_JobNumber", value);
                }
            }

            private string _PONumber = string.Empty;

            public string PONumber
            {
                get
                {
                    return _PONumber;
                }
            }

            public object SetPONumber
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_PONumber", value);
                }
            }

            private int _QuoteID = 0;

            public int QuoteID
            {
                get
                {
                    return _QuoteID;
                }
            }

            public object SetQuoteID
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_QuoteID", value);
                }
            }

            private int _ContractID = 0;

            public int ContractID
            {
                get
                {
                    return _ContractID;
                }
            }

            public object SetContractID
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_ContractID", value);
                }
            }

            private bool _ToBePartPaid = false;

            public bool ToBePartPaid
            {
                get
                {
                    return _ToBePartPaid;
                }
            }

            public object SetToBePartPaid
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_ToBePartPaid", value);
                }
            }

            private double _retention = 0;

            public double Retention
            {
                get
                {
                    return _retention;
                }
            }

            public object SetRetention
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_retention", value);
                }
            }

            private bool _CollectPayment = Conversions.ToBoolean(0);

            public bool CollectPayment
            {
                get
                {
                    return _CollectPayment;
                }
            }

            public object SetCollectPayment
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_CollectPayment", value);
                }
            }

            private bool _POC = false;

            public bool POC
            {
                get
                {
                    return _POC;
                }
            }

            public object SetPOC
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_POC", value);
                }
            }

            private bool _OTI = false;

            public bool OTI
            {
                get
                {
                    return _OTI;
                }
            }

            public object SetOTI
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_OTI", value);
                }
            }

            private bool _fOC = false;

            public bool FOC
            {
                get
                {
                    return _fOC;
                }
            }

            public object SetFOC
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_fOC", value);
                }
            }

            private int _salesRepUserID = 0;

            public int SalesRepUserID
            {
                get
                {
                    return _salesRepUserID;
                }
            }

            public object SetSalesRepUserID
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_salesRepUserID", value);
                }
            }

            private int _jobCreationType = 0;

            public int JobCreationType
            {
                get
                {
                    return _jobCreationType;
                }
            }

            public object SetJobCreationType
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_jobCreationType", value);
                }
            }

            private string _Headline = string.Empty;

            public string Headline
            {
                get
                {
                    return _Headline;
                }
            }

            public object SetHeadline
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_Headline", value);
                }
            }



            private ArrayList _assets = new ArrayList();

            public ArrayList Assets
            {
                get
                {
                    return _assets;
                }

                set
                {
                    _assets = value;
                }
            }

            private ArrayList _jobOfWorks = new ArrayList();

            public ArrayList JobOfWorks
            {
                get
                {
                    return _jobOfWorks;
                }

                set
                {
                    _jobOfWorks = value;
                }
            }

            private DataView _JobQualificationsLevels = new DataView();

            public DataView JobQualificationsLevels
            {
                get
                {
                    return _JobQualificationsLevels;
                }

                set
                {
                    _JobQualificationsLevels = value;
                    _JobQualificationsLevels.AllowEdit = true;
                    _JobQualificationsLevels.AllowNew = false;
                    _JobQualificationsLevels.AllowDelete = false;
                }
            }

            private DataView _JobSheets = new DataView();

            public DataView JobSheets
            {
                get
                {
                    return _JobSheets;
                }

                set
                {
                    _JobSheets = value;
                    _JobSheets.AllowEdit = true;
                    _JobSheets.AllowNew = false;
                    _JobSheets.AllowDelete = false;
                }
            }


        }
    }
}